src/OrleansRuntime/Catalog/Catalog.cs

```diff
@@ -1259,6 +1259,8 @@ private Task DestroyActivations(List<ActivationData> list)
                                 grainCreator.Release(activationData, grainInstance);
                             }
                         }
+
+                        activationData.Dispose();
```

Comment:
Sure, moved to a separate try block that includes both grain release (which should in theory NEVER throw otherwise the infrastructure has a nasty bug).
Disposing the objects in the scoped container shouldn't throw either, but because it's userland, I wouldn't just blindly trust that it will never throw :)